IRC.md

```diff
@@ -1,111 +1,232 @@
+
+- If you send more than 20 commands or messages to the server within a 30 second period, you will be locked out for 8 hours automatically. These are **not** lifted so please be careful when working with IRC!
+- This limit is elevated to 100 messages per 30 seconds for users that *only* send messages/commands to channels in which they have Moderator/Operator status.
 
 ## Commands you can send
+
 If you send an invalid command, you will receive a 421 numeric back:
+
 ```
-< CAP REQ :multi-prefix
-> :tmi.twitch.tv 421 you CAP :Unknown command
+< WHO #channel
+> :tmi.twitch.tv 421 twitch_username WHO :Unknown command
 ```
 
 A brief list of commands supported by our IRC server include:
-### JOIN: Opening up a chat room (and obtaining a list of users)
-**JOIN** *#channelname*
+### JOIN: Opening up a chat room
 
-Notes:
+**JOIN** *#channel*
 
-After a successful JOIN, the following will take place:
-
-1. You will be sent a list of users that are currently in the channel.
-2. A number of MODEs will be set from the **jtv** user to signify that a user can moderate chat. Users that can moderate chat are are defined as *Channel Moderators*, *Admin*'s and *Staff*.
+Note: After a successful `JOIN`, you will *not* receive a membership state events (`NAMES`, `JOIN`,`PART`, or `MODE`) *unless* you've requested our [IRCv3 `Membership`](#membership) capability
 
 ```
-< JOIN #channelname
-> JOIN #channelname
-> 353: = #channelname nickname nickname2 nickname3 nickname4 anotherNickname
-> 353: = #channelname nickname25 nickname26 nicknameN
-> 366: #channelname End of /NAMES list
-> jtv MODE #channelname +o channel_moderator
-> jtv MODE #channelname +o channel_moderator2
-> jtv MODE #channelname +o staff_user
-> jtv MODE #channelname +o twitch_global_mod_user
+< JOIN #channel
+> :twitch_username!twitch_username@twitch_username.tmi.twitch.tv JOIN #channel
+> :twitch_username.tmi.twitch.tv 353 twitch_username = #channel :twitch_username
+> :twitch_username.tmi.twitch.tv 366 twitch_username #channel :End of /NAMES list
 ```
+
 ### PART: Leaving a chat room
-**PART** *#channelname*
+
+**PART** *#channel*
+
 ```
-< PART #channelname
-> PART #channelname
+< PART #channel
+> :twitch_username!twitch_username@twitch_username.tmi.twitch.tv PART #channel
 ````
-### WHO: Obtaining a detailed list of users
-**WHO** *#channel*
+
+### PRIVMSG: Sending a message
+
+**PRIVMSG** *#channel* :Message to send
+
+Basic message reception is as follows:
+
+```
+> :twitch_username!twitch_username@twitch_username.tmi.twitch.tv PRIVMSG #channel :message here
+```
+
+# Twitch Capabilities
+
+Using IRCv3 capability registration, it is possible to register for Twitch-specific
+capabilities. The capabilities are defined below:
+
+## Membership
+
+    < CAP REQ :twitch.tv/membership
+    > :tmi.twitch.tv CAP * ACK :twitch.tv/membership
+
+Adds JOIN, PART, NAMES, and MODE functionality. By default we do *not* send this data to clients without this capability.
```

Comment:
Please use code format for these, just like above. Ideally they'd even be in the same order.